tests/test_vec_envs.py

```diff
@@ -120,6 +120,68 @@ def make_env():
     vec_env.close()
 
 
+STEP_ENV_MAX_STEPS = 5
+
+
+class StepEnv(gym.Env):
+    def __init__(self):
+        """Gym environment for testing that terminal observation is inserted
+        correctly."""
+        self.action_space = gym.spaces.Discrete(2)
+        self.observation_space = gym.spaces.Box(np.array([0]), np.array([999]),
+                                                dtype='int')
+        self.max_steps = STEP_ENV_MAX_STEPS
+        self.current_step = 0
+
+    def reset(self):
+        self.current_step = 0
+        return np.array([self.current_step], dtype='int')
+
+    def step(self, action):
+        prev_step = self.current_step
+        self.current_step += 1
+        done = self.current_step >= self.max_steps
+        # return current step as observation
+        return np.array([prev_step], dtype='int'), 0.0, done, {}
+
+
+@pytest.mark.parametrize('vec_env_class', VEC_ENV_CLASSES)
+@pytest.mark.parametrize('vec_env_wrapper', VEC_ENV_WRAPPERS)
+def test_vecenv_terminal_obs(vec_env_class, vec_env_wrapper):
+    """Test that 'terminal_observation' gets added to info dict upon
+    termination."""
+    vec_env = vec_env_class([StepEnv for _ in range(N_ENVS)])
+
+    if vec_env_wrapper is not None:
+        if vec_env_wrapper == VecFrameStack:
+            vec_env = vec_env_wrapper(vec_env, n_stack=2)
+        else:
+            vec_env = vec_env_wrapper(vec_env)
+
+    zero_acts = np.zeros((N_ENVS,), dtype='int')
+    prev_obs_b = vec_env.reset()
+    for step_num in range(STEP_ENV_MAX_STEPS - 1):
+        prev_obs_b, _, dones, infos = vec_env.step(zero_acts)
+        assert not np.any(dones)
+        assert not any('terminal_observation' in info_dict for info_dict in infos)
+
+    terminal_obs_b, _, dones, infos = vec_env.step(zero_acts)
```

Comment:
Can we make `StepEnv` terminate after different numbers of steps (can make it a constructor argument and partially apply the constructor when building `vec_env`)? Since all the environments are the same right now this wouldn't catch bugs where the terminal observation gets stored in the wrong info dict.